qiskit/circuit/library/evolved_operator_ansatz.py

```diff
@@ -0,0 +1,223 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The evolved operator ansatz."""
+
+from typing import List, Union, Optional
+
+import numpy as np
+
+from qiskit.circuit import Parameter, ParameterVector, QuantumRegister, QuantumCircuit
+from qiskit.circuit.exceptions import CircuitError
+
+from .blueprintcircuit import BlueprintCircuit
+
+
+class EvolvedOperatorAnsatz(BlueprintCircuit):
+    """The evolved operator ansatz."""
+
+    def __init__(
+        self,
+        operators: Optional[
+            Union["OperatorBase", QuantumCircuit, List[Union["OperatorBase", QuantumCircuit]]]
+        ] = None,
+        reps: int = 1,
+        evolution: Optional["EvolutionBase"] = None,
+        insert_barriers: bool = False,
+        name: str = "EvolvedOps",
```

Comment:
Just a quick question: why not just `"EvolvedOp"`?

Reply:
because it can be a list of operators 😄